core/pkg/ingress/types.go

```diff
@@ -40,6 +41,10 @@ var (
 // Controller holds the methods to handle an Ingress backend
 // TODO (#18): Make sure this is sufficiently supportive of other backends.
 type Controller interface {
+	// HealthzChecker returns is a named healthz check that returns the ingress
+	// controller status
+	healthz.HealthzChecker
```

Comment:
If the only purpose of the healthzChecker is to have a Check method, it should be renamed to healthChecker. If that is not the case (i.e healthzChecker is an interface for especially *http* health checks), we shouldn't embed it, because adding methods to healthzChecker will directly bubble up here. Reading the interface definition one could get tricked into thinking that all backends must return a http error from the Check() call, when that's not the case. The inclusion of `z` in the name leads to unnecessary confusion IMO.